### PR TITLE
take fee on approval of a milestone

### PR DESCRIPTION
**Pull request by f-gate**
This completes the [Take fee on approval task](https://app.clickup.com/t/3rcdkhj).

The system will take a fee per milestone approval. 
For example if the raised funds is 100_000, the milestone is 50%, and the fee is 1% then the calculation is:
``` 100_000 * 0.5 * 0.01```

Refunds should be untouched as they only apply to unapproved milestones (therefore a fee has not been taken).

Possible issues that need review: 
- We are not using an up to date chainstate which means that the treasury account id is not funded when benchmarks are run.
- This means that small amounts of raised funds error because the existential deposit is greater than the funds transferred to the empty treasury account. Ensuring that the treasury account is funded will make this a non issue, perhaps we can add this for benchmarking.
- Due to the rounding down of calculations, we may be left with some dust and at the moment we dont handle it at all, I am thinking some kind of OnUnbalanced trait to deal with this where appropriate.

TLDR issues: 
- Use up to date chainstate.
- handle dust after calculations.

If everyones ok i will add these to clickup.

Todo: Migration for project  - [ ]

### CODE REVIEW

**Comment by f-gate** [comment]
NEED TO DO MIGRATION

**Comment by samelamin** [comment]
Approved, would you like to do the migration? If not I am happy pushing a migration to this branch

**Comment by f-gate** [comment]
I would like to this time as it will be my first migration!

**Comment by f-gate** [comment]
Closed as per new designs
